distribution/lib/Standard/Visualization/0.0.0-dev/src/Table/Visualization.enso

```diff
@@ -187,21 +178,22 @@ make_json_for_xml_element xml_element max_items type:Text="XML_Element" =
      to display.
    - all_rows_count: the number of all rows in the underlying data, useful if
      only a fragment is displayed.
-make_json_for_table : Table -> Integer -> Boolean -> Boolean -> JS_Object
-make_json_for_table dataframe all_rows_count include_index_col is_db_table =
-    get_vector c = Warning.set (c.to_vector.map v-> make_json_for_value v) []
+make_json_for_table : Table -> Integer -> Integer -> Boolean -> Boolean -> JS_Object
+make_json_for_table dataframe max_rows all_rows_count include_index_col is_db_table =
+    act_max = if max_rows < all_rows_count then max_rows else all_rows_count
+    get_vector c = Warning.set (Vector.new act_max i-> make_json_for_value (c.get i)) []
     columns     = dataframe.columns
     header      = ["header", columns.map .name]
     value_type  = ["value_type", columns.map .value_type]
     data        = ["data",   columns.map get_vector]
     all_rows    = ["all_rows_count", all_rows_count]
     has_index_col = ["has_index_col", include_index_col]
     links       = ["get_child_node_action", "get_row"]
-    number_of_nothing = if is_db_table then Nothing else columns.map c-> c.count_nothing
-    number_of_whitespace= if is_db_table then Nothing else columns.map c-> whitespace_count c
-    nothing_p = JS_Object.from_pairs [["name", "Number of nothings"], ["percentage_value", number_of_nothing]]
-    whitespace_p = JS_Object.from_pairs [["name", "Number of untrimmed whitespace"], ["percentage_value",number_of_whitespace]]
-    data_quality_metrics =  [nothing_p, whitespace_p]
+    data_quality_metrics = if is_db_table then [] else
+        number_nothing = JS_Object.from_pairs [["name", "Count nothings"], ["percentage_value", columns.map .count_nothing]]
+        number_untrimmed = JS_Object.from_pairs [["name", "Count untrimmed whitespace"], ["percentage_value", columns.map c-> if c.length > 10000 then Nothing else c.count_untrimmed]]
+        number_untrimmed_sampled = JS_Object.from_pairs [["name", "Count untrimmed whitespace (sampled)"], ["percentage_value", columns.map c-> if c.length <= 10000 then Nothing else c.count_untrimmed]]
+        [number_nothing, number_untrimmed, number_untrimmed_sampled]
```

Comment:
nitpick, feel free to ignore, but I'd rather merge the unrimmed sample/full into one piece. 
Currently you are e.g. repeating the constant 10000 twice which may bite us if we change the threshold and forget to update it in both places.

```suggestion
        # multiline if would be useful here...
        number_untrimmed = case all_rows_count > 10000 of
            True ->
                JS_Object.from_pairs [["name", "Count untrimmed whitespace (sampled)"], ["percentage_value", columns.map c-> c.count_untrimmed]]

            False ->
                JS_Object.from_pairs [["name", "Count untrimmed whitespace"], ["percentage_value", columns.map c-> c.count_untrimmed]]
        [number_nothing, number_untrimmed]
```

Ooops having refactore this - I see both branches are almost identical (in original code too) - where is the sampling taking place??

----
 
Also looking at this sample, I feel like it may be better to have separate sets of metrics per each column, instead of sending all metrics for all columns in the whole table and having to set them to `null`. But that's for later iterations as probably not worth changing soon before release.

Reply:
Ah, now I realize that there is no difference in the branches because the decision whether to sample is done by `countUntrimmed` itself.

Well, in that case perhaps the name of the metric maybe should come from that `Column` as well? But we can tackle it as part of some future refactoring where we move metrics closer to columns/storages.